src/js/media-controller.js

```diff
@@ -36,6 +36,25 @@ class MediaController extends MediaContainer {
     // Track externally associated control elements
     this.associatedElements = [];
 
+    const mutationCallback = (mutationsList, _observer) => {
+      const { addedNodes, removedNodes } = toNextMediaChromeNodesState(mutationsList);
+      addedNodes.forEach(this.associateElement.bind(this));
+      removedNodes.forEach(this.unassociateElement.bind(this));
+    };
+
+    const observer = new MutationObserver(mutationCallback);
+    observer.observe(this, { childList: true, attributes: true, subtree: true });
```

Comment:
Yup. This is part of the "option 1" approach. Although it fires a lot, we ignore most of them without doing too much JS heavy lifting, see:
https://github.com/muxinc/media-chrome/pull/44/files#diff-f093c2ba0a65c5a566a4ad86659ffc5bee29bfbbc241f863bed8f634b7a0f9f7R462
Haven't run a perf on it, but I suspect it'll have trivial impact even for media-controls near the root of a large DOM.

The biggest pro is that we can reliably support a couple of scenarios with this approach:
- HTML only components
- Components that dynamically change whether they should be provided media chrome attrs and which attrs they should be provided
  - This also makes it less likely component implementors will accidentally fail to "unassociate" their media chrome elements

That said, we could put the onus on the component implementors to just dispatch the newly added event(s) when they want to "associate"/"unassociate", which solves the same generic problem in a different way.

Reply:
Or we say that the attribute needs to be there when the child is attached, so that the childlist change can pick it up. Not sure how often people will want to change that attr but I get the completeness of it. I still worry a little about the performance of it since we're intentionally changing a crap ton of attributes in this architecture, but I'm fine waiting to see if it's actually an issue.